modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/AnnotationProcessorUtils.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.capability.xml.schema;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.lang.annotation.Annotation;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.util.ElementFilter;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Utility class for annotation processing using the {@link javax.annotation.processing.Processor} API
+ *
+ * @since 3.7.0
+ */
+final class AnnotationProcessorUtils
```

Comment:
This class seems to be doing some hard work, it would be better to have an interface which this class can implement so classes using this can be tested in isolation